python/dgl/transform.py

```diff
@@ -1315,9 +1398,16 @@ def to_simple(g, return_counts='count', writeback_mapping=None):
         for count, canonical_etype in zip(counts, g.canonical_etypes):
             simple_graph.edges[canonical_etype].data[return_counts] = count
 
-    if writeback_mapping is not None:
+    if share_ndata:
+        for ntype in g.ntypes:
+            for key in g.nodes[ntype].data:
+                simple_graph.nodes[ntype].data[key] = g.nodes[ntype].data[key]
+
+    if writeback_mapping:
+        wb_map = {}
         for edge_map, canonical_etype in zip(edge_maps, g.canonical_etypes):
-            g.edges[canonical_etype].data[writeback_mapping] = edge_map
+            wb_map[canonical_etype] = edge_map
+        return simple_graph, wb_map
```

Comment:
In the case of graphs with a single edge type, this should return a tensor rather than a dict of tensor.